src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/blocklist.component.js

```diff
@@ -0,0 +1,536 @@
+(function () {
+    "use strict";
+
+
+    /**
+     * @ngdoc component
+     * @name Umbraco.Editors.BlockList.blockListPropertyEditor
```

Comment:
This isn't really the name though ? In other components/directives the name is somethnig like: `umbraco.directives.directive:umbButton` Also the capitalization of Umbraco, Editors, BlockList doesn't seem right since i don't think we use that anywhere in JS? I'm not sure what is correct here but this doesn't seem consistent and there's a few other components in this PR that are similar

Reply:
I had a hard time figuring out the system, since nearly no other Property Editors have JSDocs.
The "Umbraco.Web.UI.Client/src/views/propertyeditors/readonlyvalue/readonlyvalue.controller.js" has the same naming convention but would make sense to correct it if it's wrong. There is also a lot of directives that are components now, should we then update the naming to correspond that? I'm not completely sure what impact the name has?

Reply:
@nielslyngsoe cool but in the case of not figuring out the system, instead of just inventing something and hoping for the best you should hopefully ask someone on the team if they know or do the research to find out, or write a TODO or create a task to find out. I don't really know since this isn't my expertise but a quick googling I found this SO article: https://stackoverflow.com/questions/39189658/how-to-annotate-ngdoc-for-an-angular-component

We don't normally document any of the controllers since they are not normally publicly consumable and we don't want them to be. A component is a directive, they are more or less the same and they exist within a module, in this case its explicitly set to be in the "umbraco" module so based on other directive docs, it should follow the same naming procedure. There's quite a few ngdoc that i think need to be fixed up.

Reply:
Ah, now I see that you are also referring to he ngdoc type, well that's just a shift mistake. Corrected now.